doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,565 @@
+            uint32_t counter = sample->counter;
+        })
+        .if_empty([] { /* no data received but also no error */ })
+        .or_else([](iox::popo::ChunkReceiveError) { /* handle the error */ });
+}
+
+```
+
+By calling ``take`` we get a ``iox::cxx::expected<iox::optional<iox::popo::Sample<const CounterTopic>>>``. Since this
+may fail, we handle a potential error in the ``or_else`` branch. If we wake up periodically, it is also possible that
+no data is received and if we want to handle this we can optionally do so in the ``if_empty`` branch.
+
+The usual case is that we actually receive data, and we process it in the ``and_then``. Notice that in the lambda we
+do not pass a ``CounterTopic`` directly but a reference to a ``iox::popo::Sample<const CounterTopic>&``. We can
+access the underlying ``CounterTopic`` either by getting a pointer to it via ``get`` or by using ``operator->``. In
+any case, we now can process or copy the received data and once the ``sample`` goes out of scope, the underlying
+``CounterTopic`` object is deleted as well (this happens when the temporary object returned by ``take`` is
+destroyed). This means it is only safe to hold references to the data as long as the ``sample`` exists. Should we
+need a longer lifetime, we have to copy or move the data from the ``sample``.
+
+This only allows us to get one sample at a time. Should we want to get all currently available samples we can do so
+by using an additional loop.
+
+```cpp
+while (keepRunning)
+{
+    while (subscriber.hasNewSamples())
+    {
+        subscriber->take()
+            .and_then([](iox::popo::Sample<const CounterTopic>& sample) {
+                CounterTopic* ptr = sample.get();
+                /* process the received data using the ptr */
+
+                /* alternatively use operator-> */
+                uint32_t counter = sample->counter;
+            })
+            .or_else([](iox::popo::ChunkReceiveError) { /* handle the error */ });
+    }
+    // wait for new samples (either sleep or use the waitset)
+}
+```
+
+Here we do not check whether we actually have data since we already know there is data available by calling
+``hasNewSamples``.
+
+### Untyped API
+
+The untyped API offeres similar capabilities and is hence usable in a similar way. The major difference is that
+neither publisher nor subscriber have any knowledge about the underlying type they send or receive. This means that
+the user is responsible to ensure the data is read correctly, i.e. there is no type safety guaranteed by the API
+itself.
+
+#### Creating a publisher
+
+When creating an untyped publisher we do not need to specify a data type as template paraemter.
+
+```cpp
+iox::popo::UntypedPublisher publisher({"Group", "Instance", "CounterTopic"});
+publisher.offer();
+```
+
+#### Sending data
+
+1. Loan and emplace
+
+    Before sending, we have to loan a chunk of memory to emplace our data.
+
+    ```cpp
+    auto result = publisher.loan(sizeof(CounterTopic));
```

Comment:
Can introduce one, sure but i am not sure whether this will just bloat the code. Ultimately you will want to construct some data to such an array (could mean copy data there as well).

Reply:
On second thought, no I will leave it as is. The idea is to illustrate how you can achieve the same thing compared to the typed API with the untyped one. The transfer to a raw buffer and size should be pretty obvious in my book. Anyone not seeing it has no business using te untyped API;-)